Lagrange.OneBot/Core/Network/Service/HttpPostService.cs

```diff
@@ -99,12 +96,25 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     private async Task HeartbeatLoop(CancellationToken token)
     {
         var interval = TimeSpan.FromMilliseconds(_options.HeartBeatInterval);
+        Stopwatch sw = new();
+
+        if (!_options.HeartBeatEnable || _options.HeartBeatInterval <= 0)
+        {
+            await Task.Delay(Timeout.Infinite, token);
+        }
+
         while (true)
         {
             var status = new OneBotStatus(true, true);
             var heartBeat = new OneBotHeartBeat(context.BotUin, (int)_options.HeartBeatInterval, status);
+
+            sw.Start();
             await SendJsonAsync(heartBeat, null, token);
-            await Task.Delay(interval, token);
+            sw.Stop();
+
+            // Implementing precise intervals by subtracting Stopwatch's timing from configured intervals
+            await Task.Delay(interval - sw.Elapsed, token);
```

Comment:
maybe check whether `interval - elapsed` is greater than 0

Reply:
https://github.com/KonataDev/Lagrange.Core/pull/303/commits/3208249a99702004eae476847f6c77550681c24e